### PR TITLE
Refactor - post request

### PR DESCRIPTION
**Pull request by polikas**
modified the model to handle if the author exists only then add comment and added a couple of test to further test endpoint edge cases

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -232,6 +232,18 @@ describe('POST', () => {
                 expect(body.message).toBe('Bad Request');
             })
         })
+        test('status 400 Bad Request message due to invalid data', () => {
+            return request(app)
+            .post('/api/articles/1432742/comments')
```
Comment (pippaamy) [inline __tests__/app.test.js:237]: I would have this as a valid article id as testing for no body with comment :)

**Review by pippaamy — COMMENTED** [review]
Great job!
